### PR TITLE
[CWS] remove reflection in CIDR evaluator

### PR DESCRIPTION
**Pull request by safchain**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: docs/cloud-workload-security/agent_expressions.md [hunk 1]
```diff
@@ -119,6 +125,17 @@ List of the available variables:
 |-----------------------|---------------------------------------|---------------|
 | `process.pid`         | Process PID                           | 7.33          |
 
+## CIDR and IP range
+CIDR and IP matching is possible in SECL. One can use operator such as `in`, `not in`, `allin` combined with CIDR or IP notations.
```
Comment (urseberry) [inline docs/cloud-workload-security/agent_expressions.md:129]:
```suggestion
CIDR and IP matching is possible in SECL. One can use operators such as `in`, `not in`, or `allin` combined with CIDR or IP notations.
```

File: docs/cloud-workload-security/scripts/templates/agent_expressions.md [hunk 2]
```diff
@@ -105,6 +111,18 @@ List of the available variables:
 |-----------------------|---------------------------------------|---------------|
 | `process.pid`         | Process PID                           | 7.33          |
 
+## CIDR and IP range
+CIDR and IP matching is possible in SECL. One can use operator such as `in`, `not in`, `allin` combined with CIDR or IP notations.
```
Comment (urseberry) [inline docs/cloud-workload-security/scripts/templates/agent_expressions.md:115]:
```suggestion
CIDR and IP matching is possible in SECL. One can use operators such as `in`, `not in`, or `allin` combined with CIDR or IP notations.
```

**Review by urseberry — APPROVED** [review]

File: docs/cloud-workload-security/agent_expressions.md [hunk 3]
```diff
@@ -79,6 +79,12 @@ SECL operators are used to combine event attributes together into a full express
 | `\|`                  | File             | Binary or                                | 7.27          |
 | `&&`                  | File             | Logical and                              | 7.27          |
 | `\|\|`                | File             | Logical or                               | 7.27          |
+| `in CIDR`             | Network          | Element is in the IP range               | 7.36          |
+| `not in CIDR`         | Network          | Element is not in the IP range           | 7.36          |
+| `allin CIDR`          | Network          | All the elements are in the IP range     | 7.36          |
```
Comment (lebauce) [inline docs/cloud-workload-security/agent_expressions.md:84]: 7.37

File: pkg/security/secl/compiler/ast/secl.go [hunk 4]
```diff
@@ -158,7 +158,7 @@ type ScalarComparison struct {
 type ArrayComparison struct {
 	Pos lexer.Position
 
-	Op    *string `parser:"( @( \"in\" | \"not\" \"in\" | \"intersects\" )"`
+	Op    *string `parser:"( @( \"in\" | \"not\" \"in\" | \"allin\" )"`
```
Comment (lebauce) [inline pkg/security/secl/compiler/ast/secl.go:161]: Do you think we should implement "allin" for something else than CIDR ? (like [ "XXX", "YYY" ] allin process.capabilities ?

Comment (safchain) [reply pkg/security/secl/compiler/ast/secl.go:161]: Probably, let's add the in a following PR

**Review by paulcacheux — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

**Pull request merged**